apps/designer/app/canvas/shared/styles.ts

```diff
@@ -0,0 +1,84 @@
+import store from "immerhin";
+import { useSubscribe } from "~/shared/pubsub";
+import { setInstanceStyleMutable } from "~/shared/tree-utils";
+import { useSelectedInstance } from "./nano-states";
+import { rootInstanceContainer } from "~/shared/nano-states";
+import {
+  type StyleValue,
+  toValue,
+  toVarNamespace,
+} from "@webstudio-is/react-sdk";
+import { globalCss } from "@webstudio-is/design-system";
+
+const setCssVar = (id: string, property: string, value?: StyleValue) => {
+  const customProperty = `--${toVarNamespace(id, property)}`;
+  if (value === undefined) {
+    document.body.style.removeProperty(customProperty);
+    return;
+  }
+  document.body.style.setProperty(customProperty, toValue(value));
+};
+
+const useUpdateStyle = () => {
+  const [selectedInstance] = useSelectedInstance();
+  useSubscribe("updateStyle", ({ id, updates, breakpoint }) => {
+    // Only update styles if they match the selected instance
+    // It can potentially happen that we selected a difference instance right after we changed the style in style panel.
+    if (id !== selectedInstance?.id) return;
+
+    for (const update of updates) {
+      setCssVar(id, update.property, undefined);
+    }
```

Comment:
Seem strange that we pass `undefined` as value, should it be something like `update.value`?

Reply:
its deletion operation, or basically an "unset", worth rewriting into a separate semantic?

Reply:
Nah, just wanted to make sure this is not a mistake